py/torch_tensorrt/dynamo/aten_tracer.py

```diff
@@ -20,13 +23,54 @@ def trace(
     # Set log level at the top of compilation (torch_tensorrt.dynamo)
     if ("debug" in kwargs and kwargs["debug"]) and logger.parent:
         logger.parent.setLevel(logging.DEBUG)
+
+    # Determine the dynamic dimension and setup constraints to input dimensions as dictated by TensorRT
+    # Torch dynamo does not allow 0/1 value for dynamic dimensions
+    # for inputs during tracing. Hence we create new inputs for export
+    device = to_torch_device(kwargs.get("device", default_device()))
+    torch_inputs = get_torch_inputs(inputs, device)
+    trace_inputs = []
+    constraints = []
+    for idx, input in enumerate(inputs):
+        if input.shape_mode == Input._ShapeMode.DYNAMIC:
+            min_shape = input.shape["min_shape"]
+            opt_shape = input.shape["opt_shape"]
+            max_shape = input.shape["max_shape"]
+            assert len(min_shape) == len(opt_shape) == len(max_shape)
+
+            constraint_dims = []
+            new_shape = []
+            for dim in range(len(min_shape)):
+                if min_shape[dim] == opt_shape[dim] == max_shape[dim]:
+                    new_shape.append(torch_inputs[idx].shape[dim])
+                else:
+                    constraint_dims.append(dim)
+                    if torch_inputs[idx].shape[dim] == 1:
+                        new_shape.append(torch_inputs[idx].shape[dim] + 1)
```

Comment:
Given that we are not using the `SymInt` approach here and a trace is made for each shape, why do we need to consider Torch's 0/1 specialization restriction in this case?

Reply:
Not sure, I understood the question completely but here's what's going on.
1) torch.export(model, inputs) returns a model which only works for given input shape (model(inputs) will fail if there's a different input shape eg: different batch too). So, we use contraints API provided by Torch which inserts `sym_int` ops in the graph which operate on the dynamic dimension. 
2) Torch's 0/1 specialization is due to their implementation on dynamo. In the case of dynamic shapes, we can't have the tracing inputs to have `1` in the dynamic dimension and this is a restriction (and not a bug) as per Angela's comments from Meta.
Does that answer your question ?

Reply:
This does answer my question, yes - thank you for the detailed response. How do we "correct" for this off-by-one issue then, if the user specified a minimum size of 1?

Reply:
I detect the dynamic dimension to be 1 and in this case, I just add 1 and create a new tensor with it. https://github.com/pytorch/TensorRT/blob/16d19e80e89a9538d2f404bc39b3a88b1717da1b/py/torch_tensorrt/dynamo/aten_tracer.py#L49